rkt: create parent cgroup to enable stats

schmichael opened the pull request:
Fixes #2400

Having the Nomad executor create parent cgroups that rkt is launched
within allows the stats collection code used for the exec driver to Just
Work. The only downside is that now the Nomad executor's resource
utilization counts against the cgroups resource limits just as it does
for the exec driver.

dadgar commented on client/driver/rkt.go:
```diff
@@ -194,7 +194,10 @@ func rktGetDriverNetwork(uuid string, driverConfigPortMap map[string]string, log
 				}
 
-				// This is a successful landing.
+				// This is a successful landing; log if its not the first attempt.
```
Is there a test we can add to ensure stats work?

schmichael commented:
Added a test and modernized rkt tests a bit more. Also removed the env var required to run them as we already require Linux + root + rkt installed. Since it's an officially supported driver we probably shouldn't skip the tests by default.

dadgar commented on client/testutil/driver_compatible.go:
```diff
@@ -38,13 +39,24 @@ func QemuCompatible(t *testing.T) {
 func RktCompatible(t *testing.T) {
 	if runtime.GOOS == "windows" || syscall.Geteuid() != 0 {
-		t.Skip("Must be root on non-windows environments to run test")
+		t.Skip("Must be root on non-Windows environments to run test")
```
This would also try on Darwin

dadgar commented on client/testutil/driver_compatible.go:
```diff
@@ -38,13 +39,24 @@ func QemuCompatible(t *testing.T) {
+	rktOnce.Do(func() {
+		out, err := exec.Command("rkt", "version").CombinedOutput()
+		if err == nil {
+			t.Logf("found rkt version: %s", out)
```
Can we remove this log?
